crates/proof-of-sql/src/proof_primitive/hyrax/base/hyrax_helpers.rs

```diff
@@ -0,0 +1,187 @@
+    let nu = point.len() / 2 + 1;
+    let mut lo_vec = vec![S::ZERO; 1 << nu];
+    let mut hi_vec = vec![S::ZERO; 1 << nu];
+    lo_vec[0] = point.iter().take(nu).map(|b| S::ONE - *b).product();
+    hi_vec[0] = point.iter().skip(nu).map(|b| S::ONE - *b).product();
+    let standard_basis_point = point
+        .iter()
+        .map(|b| (S::ONE - *b).inv().expect("Values in point cannot be 1.") - S::ONE)
+        .collect_vec();
+    compute_dynamic_standard_basis_vecs(&standard_basis_point, &mut lo_vec, &mut hi_vec);
+    (lo_vec, hi_vec)
+}
+
+/// Returns the (row, column) in the matrix where the data with the given index belongs.
+pub(crate) const fn row_and_column_from_index(index: usize) -> (usize, usize) {
+    let width_of_row = 1 << (((2 * index + 1).ilog2() + 1) / 2);
+    let row = index / width_of_row + width_of_row / 2;
+    let column = index % width_of_row;
+    (row, column)
+}
+
+pub(crate) const fn full_width_of_row(row: usize) -> usize {
+    ((2 * row + 4) / 3).next_power_of_two()
+}
+
+pub(crate) const fn matrix_size(data_len: usize, offset: usize) -> (usize, usize) {
+    if data_len == 0 && offset == 0 {
+        return (0, 0);
+    }
+
+    let (last_row, _) = row_and_column_from_index(offset + data_len - 1);
+    let width_of_last_row = full_width_of_row(last_row);
+    (last_row + 1, width_of_last_row)
+}
+
+#[allow(dead_code)]
+/// This method produces evaluation vectors from a point. This is a helper method for generating a Vector-Matrix-Vector product in the dynamic dory scheme.
+///
+/// The ith element of the `lo_vec` is essentially the ith monomial basis element (lexicographically).
+/// The ith element of the `hi_vec` is essentially the jth monomial basis element where `j = row_start_index(i)`.
+///
+/// NOTE: the `lo_vec` and `hi_vec` are scaled by `lo_vec[0]` and `hi_vec[0]` respectively.
+/// NOTE: `lo_vec` and `hi_vec` should otherwise consist entirely of zeros in order to ensure correct output.
+pub fn compute_dynamic_standard_basis_vecs<S: Scalar>(
+    point: &[S],
+    lo_vec: &mut [S],
+    hi_vec: &mut [S],
+) {
+    let nu = point.len() / 2 + 1;
+    debug_assert_eq!(lo_vec.len(), 1 << nu);
+    debug_assert_eq!(hi_vec.len(), 1 << nu);
+    for i in 1..nu {
+        build_partial_second_half_standard_basis_vecs(
+            &point[..2 * i - 1],
+            &mut lo_vec[..1 << i],
+            &mut hi_vec[..1 << i],
+            true,
+        );
+    }
+    // Note: if we don't have the "full" point, we shouldn't fill up the last quarter because it should be all zeros.
+    build_partial_second_half_standard_basis_vecs(point, lo_vec, hi_vec, point.len() % 2 == 1);
+    // Add the most significant variable, which was not included before in order to allow simple copying to work.
+    point.iter().skip(1).enumerate().for_each(|(i, v)| {
+        let p = i / 2;
+        let o = 2 + i % 2;
+        (o << p..(o + 1) << p).for_each(|k| hi_vec[k] *= *v);
+    });
+}
```

Comment:
Why is this dead? If it will be used later, please add docs for that. If it will not be used, lets go ahead and remove it.